x-pack/plugins/observability/server/services/slo/historical_summary_client.test.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClientMock, elasticsearchServiceMock } from '@kbn/core/server/mocks';
+import moment from 'moment';
+import { oneMinute, oneMonth, thirtyDays } from './fixtures/duration';
+import { createSLO } from './fixtures/slo';
+import { DefaultHistoricalSummaryClient } from './historical_summary_client';
+
+const commonEsResponse = {
+  took: 100,
+  timed_out: false,
+  _shards: {
+    total: 0,
+    successful: 0,
+    skipped: 0,
+    failed: 0,
+  },
+  hits: {
+    hits: [],
+  },
+};
+
+const generateEsResponseForRollingSLO = (
```

Comment:
💬 The response for a rolling SLO returns two times the duration buckets, each bucket is one day summary.